dimod/generators/random.py

```diff
@@ -133,3 +133,72 @@ def randint(graph, vartype, low=0, high=1, cls=BinaryQuadraticModel,
 
     return cls.from_numpy_vectors(ldata, (irow, icol, qdata), offset, vartype,
                                   variable_order=variables)
+
+
+@graph_argument('graph')
+def ran_r(r, graph, vartype, cls=BinaryQuadraticModel, seed=None):
+    """Generate a binary quadratic model for a RANr problem.
+
+    In RANr problems all linear biases are zero and quadratic values are uniformly
+    selected integers between -r to r, excluding zero.
+
+    Args:
+        r (int):
+            Order of the RANr problem.
+
+        graph (int/tuple[nodes, edges]/:obj:`~networkx.Graph`):
+            The graph to build the binary quadratic model (BQM) for. Either an
+            integer n, interpreted as a complete graph of size n, or
+            a nodes/edges pair, or a NetworkX graph.
+
+        vartype (:class:`.Vartype`/str/set):
```

Comment:
RANr problems really only make sense for SPIN-valued BQMs. I think this should just always return SPIN.

Reply:
Done